Library/Homebrew/formula_support.rb

```diff
@@ -28,13 +28,14 @@ def valid?
 
   def to_s
     return @explanation unless @explanation.empty?
+    os_name = OS.mac? ? MacOS.os_name : 'macOS'
```

Comment:
@MikeMcQuaid I think why CI complains is that when `ENV["HOMEBREW_TEST_GENERIC_OS"]` is set, `MacOS` is not initialized but we can use `OS::Mac.os_name(ENV["HOMEBREW_OS_VERSION"])` here so that both CI and macOS should be happy.

> Alternatively, edit brew.sh to set a HOMEBREW_OS_NAME (like HOMEBREW_OS_VERSION currently) which uses the correct version and then OS.name can just use ENV["HOMEBREW_OS_NAME"]

It think this is over complicated. Basically I aim to adding the pedantry in a single place so that when macOS 10.14 is released we don't have to search everywhere to remove multiple methods and their usage.